src/test/java/com/pusher/client/channel/impl/ChannelImplTest.java

```diff
@@ -201,6 +203,36 @@ public void testUnbindWhenInUnsubscribedStateThrowsException() {
         channel.updateState(ChannelState.UNSUBSCRIBED);
         channel.unbind(EVENT_NAME, mockListener);
     }
+    @Test
+    public void testMetaDataIsExtractedFromMessageAndPassedToSingleListener() {
+        final ArgumentCaptor<PusherEvent> argument = ArgumentCaptor.forClass(PusherEvent.class);
+
+        final String eventName = "client-my-event";
+        ChannelEventListener mockListener = getEventListener();
+
+        channel = newInstance(getChannelName());
```

Comment:
Why do you redefine the`mockListeners` and `channel` here and in the other test?